### PR TITLE
Move nightly build to GH actions workflow

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
- Add GitHub Actions workflow to run nightly builds of Che
- Remove unused cico_* scripts
- Added new image to build&push during the nightly build - Che Server based on rhel.Dockerfile with tag "rhel-nightly"

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18545

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
what for we need nightly builds? Can we have only the latest build from `HEAD` branch?

**Comment by nickboldt** [comment]
@skabashnyuk so you want a respin on every commit? or a respin every day (if changes since last build) ? How often do you think we need these? I'm thinking weekly might be enough, but maybe you want it more often?

File: dockerfiles/che/rhel.Dockerfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ RUN mkdir /logs /data && \
     chmod -R g+rwX /home/user && \
     find /home/user -type d -exec chmod 777 {} \; && \
     java -version && echo -n "Server startup script in: " && \
-    find /home/user/codeready -name catalina.sh | grep -z /home/user/codeready/tomcat/bin/catalina.sh
+    find /home/user/eclipse-che -name catalina.sh | grep -z /home/user/eclipse-che/tomcat/bin/catalina.sh
```
Comment (nickboldt) [inline dockerfiles/che/rhel.Dockerfile:34]: downstream we use codeready, not eclipse-che. If we make this change in upstream version we'll have to remember to also transform it in downstream jenkinsfile/sync job. 

https://gitlab.cee.redhat.com/codeready-workspaces/crw-jenkins/-/blob/master/jobs/CRW_CI/crw-server_2.6.jenkinsfile#L271-273

Comment (benoitf) [reply dockerfiles/che/rhel.Dockerfile:34]: anyway, codeready name shouldn't be in upstream

Comment (nickboldt) [reply dockerfiles/che/rhel.Dockerfile:34]: true - see merged https://github.com/eclipse/che/pull/18679 which moves to `eclipse-che` in this file, and then we re-transform it back to `codeready` in the midstream and downstream versions.

**Comment by nickboldt** [comment]
@skabashnyuk said (on slack):

> If you ask me I think we need latest from master only
> 1 time per day @ 01:00 am

So +1 for the existing PR. I'm not fussed if it runs at 00:00 every day or 01:00... unless you specifically want it at 1am EEST, rather than whatever default TZ GH uses. 

Does anyone know if GH actions support the Jenkins approach of "H H/24 * * *" or "@daily" so that this runs at a time that the system isn't busy?

**Review by nickboldt — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Review by sparkoo — APPROVED** [review]

**Pull request merged**